usr/share/rear/build/GNU/Linux/400_copy_modules.sh

```diff
@@ -1,28 +1,148 @@
 # 400_copy_modules.sh
 #
-# copy kernel modules for Relax-and-Recover
-#
+# Collect kernel modules and copy them into the rescue/recovery system.
+
 # This file is part of Relax-and-Recover, licensed under the GNU General
 # Public License. Refer to the included COPYING for full text of license.
 
-Log "Collecting modules for kernel version $KERNEL_VERSION"
-MODFILES=(
-$( ResolveModules "$KERNEL_VERSION" "${MODULES[@]}" "${MODULES_LOAD[@]}"  )
-)
-StopIfError "Could not resolve kernel module dependancies"
+# The special user setting MODULES=( 'no_modules' ) enforces that
+# no kernel modules get included in the rescue/recovery system
+# regardless of what modules are currently loaded.
+# Test the first MODULES array element because other scripts
+# in particular rescue/GNU/Linux/240_kernel_modules.sh
+# already appended other modules to the MODULES array:
+if test "no_modules" = "$MODULES" ; then
+    LogPrint "Omit copying kernel modules (MODULES contains 'no_modules')"
+    return
+fi
+
+# As general condition the /lib/modules/$KERNEL_VERSION directory must exist:
+test "$KERNEL_VERSION" || KERNEL_VERSION="$( uname -r )"
+if ! test -d "/lib/modules/$KERNEL_VERSION" ; then
+    Error "Cannot copy kernel modules because /lib/modules/$KERNEL_VERSION does not exist"
+fi
+
+# Local functions that are 'unset' at the end of this script:
+function modinfo_filename () {
+    local module_name=$1
+    local module_filename=""
+    # Older modinfo (e.g. the one in SLES10) does not support '-k'
+    # but that old modinfo returns a zero exit code when called as 'modinfo -k ...'
+    # and shows a 'modinfo: invalid option -- k ...' message on stderr and nothing on stdout
+    # so that we need to check if we got a non-empty module filename:
+    module_filename=$( modinfo -k $KERNEL_VERSION -F filename $module_name 2>/dev/null )
```

Comment:
Didn't you note that the crappy old modinfo does not
return a non-zero exit code so that
modinfo ... || Error
would not work.

If I had just ignored that ols stuff nobody would have cared.
Now I try to somehow deal even with only crap and
all is not right.
I think I simply remove all that overcomplicated attempts
doe work around crappy tools and leave users with crapy
tools alone.